hashednode.go

```diff
@@ -31,14 +31,37 @@ import (
 )
 
 type HashedNode struct {
-	commitment  []byte
+	commitment  SerializedPointCompressed
+	serialized  SerializedPoint
 	cachedPoint *Point
 }
 
+func NewHashedNode(buf SerializedPoint) *HashedNode {
+	var h HashedNode
+	h.serialized = buf
+
+	return &h
+}
+
 func (*HashedNode) Insert([]byte, []byte, NodeResolverFn) error {
 	return errInsertIntoHash
 }
 
+func (h *HashedNode) BytesUncompressed() []byte {
+	return h.serialized
+}
+
+func (h *HashedNode) BytesCompressed() []byte {
+	if h.commitment == nil {
+		if h.cachedPoint != nil {
+			h.commitment = h.cachedPoint.Bytes()
+			return h.commitment
+		}
+		h.commitment = h.serialized.ToCompressed()
+	}
+	return h.commitment
+}
```

Comment:
OK, so here is the work we wanted to delay as much as possible.
Now _HashedNode_ is asked to return the compressed version of the point.

Our source of truth always available is `serialized` (i.e: uncompressed form of the point), but we take some strategy here:
1. If `h.commitment != nil` (i.e: we already calculated this before and have it cached), use it obviously.
2. If that isn't the case, we've never calculated `h.commitment`. Do we have `h.cachedPoint` cached? If that's the case, let's get the compressed form by getting it from the in-memory point to `Bytes()`. That's faster than our last resource (next point)
3. What if we don't have that `h.cachedPoint`? We don't have any other option than do the work of transforming from our uncompressed serialized representation (i.e: `h.serialized`) to the compressed serialization form. We do this with a new method I created in go-ipa PR `.ToCompressed()`. 

That `ToCompressed()` method is somewhat costly... if we can optimize that, there's another big opportunity to improve performance. This is the topic I mentioned in our chat that I pinged @kevaundray; we can explore that whenever he can think about it.

The TL;DR of all this comment is: 
- Delay `h.commitment` calculation as much as we can
- Use whatever opportunity we have regarding other cached values to do this computation as fast as we can.

This change was very important for the replay `go-ethereum` gains.